File: stacknqueue/rearrange-positive-and-negative-no-in-array.cpp
#include <iostream>
using namespace std;

void rearrangePositiveNegative(int array[], int size) {
    int left = 0, right = size - 1;

    while (left <= right) {
        // Find the next positive number from the left
        while (left <= right && array[left] < 0) {
            left++;
        }

        // Find the next negative number from the right
        while (left <= right && array[right] >= 0) {
            right--;
        }

        // Swap the positive number at left and negative number at right
        if (left < right) {
            swap(array[left], array[right]);
            left++;
            right--;
        }
    }
}

int main() {
    int size;
    cout << "Enter the size of the array:" << endl;
    cin >> size;

    int array[size];
    cout << "Enter the elements of the array:" << endl;
    for (int i = 0; i < size; i++) {
        cout << "Enter the element " << i + 1 << ": " << endl;
        cin >> array[i];
    }

    // Function call to rearrange positive and negative numbers
    rearrangePositiveNegative(array, size);

    // Print the updated array
    cout << "Updated Array: ";
    for (int i = 0; i < size; i++) {
        cout << array[i] << " ";
    }
    cout << endl;

    return 0;
}

File: stacknqueue/rearrange-positive-and-negative-no-in-array.py
def rearrange_positive_negative(array):
    left = 0
    right = len(array) - 1

    while left <= right:
        # Find the next positive number from the left
        while left <= right and array[left] < 0:
            left += 1

        # Find the next negative number from the right
        while left <= right and array[right] >= 0:
            right -= 1

        # Swap the positive number at left and negative number at right
        if left < right:
            array[left], array[right] = array[right], array[left]
            left += 1
            right -= 1

# Main function
if __name__ == "__main__":
    size = int(input("Enter the size of the array: "))
    array = []

    print("Enter the elements of the array:")
    for i in range(size):
        element = int(input(f"Enter the element {i + 1}: "))
        array.append(element)

    # Function call to rearrange positive and negative numbers
    rearrange_positive_negative(array)

    # Print the updated array
    print("Updated Array:", array)
